src/app/client/src/app/modules/player-helper/components/player/player.component.ts

```diff
@@ -169,4 +204,71 @@ export class PlayerComponent implements AfterViewInit, OnChanges {
       }
     }
   }
+
+  /**
+   * this method will handle play button click and turn the player into landscape
+   */
+  enablePlayer(mode: boolean) {
+    this.playerThumbnail = mode;
+    this.loadPlayer();
+  }
+
+  /** this method checks for the browser capability to be fullscreen via if-else ladder
+   * if match found, it will turn the player along will be close button into fullscreen and then
+   * rotate it to landscape mode
+   */
+  rotatePlayer() {
+    setTimeout(() => {
+      const playVideo: any = document.querySelector('#playerFullscreen');
+      try {
+        if (playVideo.requestFullscreen) {
+          playVideo.requestFullscreen();
+        } else if (playVideo.mozRequestFullScreen) { /* Firefox */
+          playVideo.mozRequestFullScreen();
+        } else if (playVideo.webkitRequestFullscreen) { /* Chrome, Safari and Opera */
+          playVideo.webkitRequestFullscreen();
+        } else if (playVideo.msRequestFullscreen) { /* IE/Edge */
+          playVideo.msRequestFullscreen();
+        }
+        screen.orientation.lock('landscape');
+        this.showCloseButton = true;
+      } catch (error) {}
+    });
+  }
+
+  /** when user clicks on close button
+   * this method will let the player to exit from fullscreen mode and
+   * 1. video thumbnail will be shown for single content
+   * 2. content-details page will be shown ( for multi-result dial-code search flow)
+   */
+  closeFullscreen() {
+    this.showCloseButton = false;
+    /** to exit the fullscreen mode */
+    if (document['exitFullscreen']) {
```

Comment:
can we use a switch case

Reply:
it is not recommended to use the switch case as, we don't know which browser supports which api for rotating as well as closing. .